src/config_design.config

```diff
@@ -0,0 +1,126 @@
+%% Cloudant_benchmark config changes design review 
+
+%% ----------------------------------------------------------------------------
+%% CONFIGURATION TEMPLATES: (Fogbugz 62351)
+%% ----------------------------------------------------------------------------
+
+{config_templates, ["basho_bench", "cloudant_http", "http_driver"]}.
+
```

Comment:
I need to think about this more. Generally the config_templates we need are tied to the drivers we are using at a given time but we wont know the drivers until we have parsed all the config params, including the nested worker_types structure. So the drivers internally know what they need but we won't know that until they are being loaded at runtime. One way to think of this is as a way of specifying what the configuration information we need for the range of drivers that we MIGHT be using (i.e. there might be more worker_types defined than referenced by workers for a given run).